gbpservice/nfp/configurator/agents/vpn.py

```diff
@@ -0,0 +1,275 @@
+import json
+import os
+import oslo_messaging as messaging
+import requests
+
+from gbpservice.nfp.configurator.agents import agent_base
+from gbpservice.nfp.configurator.lib import data_filter
+from gbpservice.nfp.configurator.lib import utils
+from gbpservice.nfp.configurator.lib import vpn_constants as const
+from gbpservice.nfp.core import main
+
+from oslo_log import log as logging
+from oslo_messaging import MessagingTimeout
+
+from neutron import context as ctxt
+
+
+LOG = logging.getLogger(__name__)
+
+
+class VpnaasRpcSender(data_filter.Filter):
+    """ RPC APIs to VPNaaS Plugin.
+    """
+    RPC_API_VERSION = '1.0'
+    target = messaging.Target(version=RPC_API_VERSION)
+
+    def __init__(self, context, sc):
+        self.context = context
+        self.sc = sc
+        self.notify = agent_base.AgentBaseNotification(sc)
+        super(VpnaasRpcSender, self).__init__(None, None)
+
+    def get_vpn_services(self, context, ids=None, filters=None):
+        """Get list of vpnservices on this host.
+        """
+        return self.call(
+            context,
+            self.make_msg('get_vpn_services', ids=ids, filters=filters))
+
+    def get_vpn_servicecontext(self, context, filters=None):
+        """Get list of vpnservice context on this host.
+           For IPSEC connections :
+                List of vpnservices -->
+                lIst of ipsec connections -->
+                ike policy & ipsec policy
+        """
+        return self.call(
+            context,
+            self.make_msg(
+                'get_vpn_servicecontext', filters=filters))
+
+    def get_ipsec_conns(self, context, filters):
+        """
+        Get list of ipsec conns with filters
+        specified.
+        """
+        return self.call(
+            context,
+            self.make_msg(
+                'get_ipsec_conns',
+                filters=filters))
+
+    def update_status(self, context, status):
+        """Update local status.
+
+        This method call updates status attribute of
+        VPNServices.
+        """
+        msg = {'receiver': const.NEUTRON,
+               'resource': const.SERVICE_TYPE,
+               'method': 'update_status',
+               'data': {'context': context,
+                        'status': status}
+               }
+        self.notify._notification(msg)
+
+    def ipsec_site_conn_deleted(self, context, resource_id):
+        """ Notify VPNaaS plugin about delete of ipsec-site-conn """
+
+        msg = {'receiver': const.NEUTRON,
+               'resource': const.SERVICE_TYPE,
+               'method': 'ipsec_site_conn_deleted',
+               'data': {'context': context,
+                        'resource_id': resource_id}
+               }
+        self.notify._notification(msg)
+
+
+class VPNaasRpcManager(agent_base.AgentBaseRPCManager):
+    """
```

Comment:
Change this comment. These are not real RPC messages. They are now coming from configurator. Explain accordingly. Keep this comment block outside.